### DISCUSSION TITLE
Spindle Sync

### DISCUSSION THREAD

**Discussion by poyraz-birusk** [grblHAL/STM32F4xx#31] [General]
Is spindle Sync implemented for STM32F4 boards? I'd like to try to use it as a electronic lead screw 😄

**Comment by terjeio** [comment]
Yes, but only tested (a while ago) with a simulator and undocumented. Currently only the [st_morho_map](https://github.com/grblHAL/STM32F4xx/blob/acc0f7bcd84de4e2882df51884e9e47c5509760e/Inc/st_morpho_map.h#L128-L139) has pin definitions for it.

It can be enabled by enabling `BOARD_MORPHO_CNC` file in [my_machine.h](https://github.com/grblHAL/STM32F4xx/blob/master/Inc/my_machine.h) and adding
`#define SPINDLE_SYNC_ENABLE 1`
somewhere in that file.

Note that the spindle pulse pin has to be routed as clock input to a timer and currently only PD2 is supported. If you want to add your own pin mapping you have to use PD2 or change the code.

**Comment by nickshl** [comment]
Hi, @terjeio, by some reason PID tuning doesn't work for me. It used to work before. But now by some reason after I run program and press "Get data" button nothing show up except dashed zero line. I tried Edge and released version of ioSender, I also tried some old firmware that was stored on my laptop. Nothing works. Any ideas what I have to look for?

**Reply by terjeio** [reply]
Do you have [logging enabled](https://github.com/grblHAL/core/blob/04a5edcf8353f7294167d12d32e4b073d9be9ce4/config.h#L521-L522)?

**Reply by nickshl** [reply]
I enabled "Spindle sync" and "Log for tuning" in Web Builder. I believe, if "Log for tuning" isn't enabled, "PID Tuner" tab isn't appear in ioSender. I could try to build firmware myself later today and check again.

**Reply by nickshl** [reply]
@terjeio, I can confirm - when I build firmware myself(_I also set PID_LOG to 2048 for no reason, I didn't tried 1000_) it works. I also tried to set PID_LOG  to 1 and guess what? I got same result as with Web Builder! At this point I can speculate, that when "Log for tuning" is enabled in Web Builder it set PID_LOG to **1** to enable it. May be make sense instead tick mark "Log for tuning" place numeric field to set number of log records in Web Builder? And 0 will be mean no log for tuning.
By the way, why default is 1000, not 1024? :)

**Comment by nickshl** [comment]
Hi @terjeio, have you check which value for PID_LOG Web Builder set when "Log for tuning" is enabled? If my theory above is correct, could you fix that?

**Reply by terjeio** [reply]
It will be fixed when I commit the next grblHAL build which needs an updated Web Builder. Tomorrow?

**Reply by nickshl** [reply]
Works for me, thank you!
